src/main/java/com/minecolonies/coremod/blocks/BlockShingleSlab.java

```diff
@@ -0,0 +1,261 @@
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.CURVED).withProperty(FACING, EnumFacing.EAST);
+            }
+            else if(!connectors[0] && connectors[1] && !connectors[2] && connectors[3])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.CURVED).withProperty(FACING, EnumFacing.SOUTH);
+            }
+            if(connectors[0] && !connectors[1] && connectors[2] && !connectors[3])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.CURVED).withProperty(FACING, EnumFacing.NORTH);
+            }
+        }
+        else if(amount == THREE_CONNECTIONS)
+        {
+            if (!connectors[0])
+            {
+                if(!world.isAirBlock(position.west().down()))
+                {
+                    return state.withProperty(VARIANT, ShingleSlabType.THREE_WAY).withProperty(FACING, EnumFacing.NORTH);
+                }
+                return state.withProperty(VARIANT, ShingleSlabType.TWO_WAY).withProperty(FACING, EnumFacing.NORTH);
+            }
+            else if (!connectors[1])
+            {
+                if(!world.isAirBlock(position.east().down()))
+                {
+                    return state.withProperty(VARIANT, ShingleSlabType.THREE_WAY).withProperty(FACING, EnumFacing.SOUTH);
+                }
+                return state.withProperty(VARIANT, ShingleSlabType.TWO_WAY).withProperty(FACING, EnumFacing.SOUTH);
+            }
+            else if (!connectors[2])
+            {
+                if(!world.isAirBlock(position.south().down()))
+                {
+                    return state.withProperty(VARIANT, ShingleSlabType.THREE_WAY).withProperty(FACING, EnumFacing.WEST);
+                }
+                return state.withProperty(VARIANT, ShingleSlabType.TWO_WAY).withProperty(FACING, EnumFacing.WEST);
+            }
+            else
+            {
+                if(!world.isAirBlock(position.north().down()))
+                {
+                    return state.withProperty(VARIANT, ShingleSlabType.THREE_WAY).withProperty(FACING, EnumFacing.EAST);
+                }
+                return state.withProperty(VARIANT, ShingleSlabType.TWO_WAY).withProperty(FACING, EnumFacing.EAST);
+            }
+        }
+        return state.withProperty(VARIANT, ShingleSlabType.FOUR_WAY);
+    }
+
+    @Override
+    protected BlockStateContainer createBlockState()
+    {
+        return new BlockStateContainer(this, new IProperty[] {FACING, VARIANT});
+    }
+
+    /**
+     * @deprecated remove when minecraft invents something better.
+     */
+    @Deprecated
+    @Override
+    public boolean isOpaqueCube(@NotNull final IBlockState state)
+    {
+        return false;
+    }
+
+    /**
+     * @deprecated remove when minecraft invents something better.
+     */
+    @Deprecated
+    @Override
+    public boolean isFullCube(@NotNull final IBlockState state)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)